### PR TITLE
I99 odgi bin nuc pos

### PR DESCRIPTION
**Pull request by subwaystation**
This PR adds the detailed nucleotide position for each bin.

### CODE REVIEW

**Comment by subwaystation** [comment]
I only modified the JSON output. 

The .tsv output will still be the same, so it does not break scripts in R, etc.

@ekg Please merge, if you think it's fine. Thanks!

**Comment by josiahseaman** [comment]
Have you profiled the change in run speed and output size?  I had almost identical code in odgi earlier and I stripped it out because it seemed we only needed sign post nucleotides (first, last, median), not every single one. I probably missed a conversation. Which feature are you trying to support on the front end?

**Comment by subwaystation** [comment]
Sorry, I should have specified that. So in order to bring in annotation in our view, we either need a `bin -> node_id` mapping for each path or a `bin -> nuc_pos` mapping. I discussed this with @JervenBolleman . Else we would not know, which feature corresponds to which position. 
If someone is aware of a better solution, I am happy to go down another road.

File: src/algorithms/bin_path_info.cpp [hunk 1]
```diff
@@ -71,10 +71,13 @@ void bin_path_info(const PathHandleGraph& graph,
                         }
                         bins[curr_bin].mean_pos += path_pos++;
                         nucleotide_count += 1;
-                        if(bins[curr_bin].first_nucleotide == 0){
-                            bins[curr_bin].first_nucleotide = nucleotide_count;
-                        }
-                        bins[curr_bin].last_nucleotide = nucleotide_count;
+						if((bins[curr_bin].ranges.size() == 0) ||
+						    ((nucleotide_count -  bins[curr_bin].ranges[bins[curr_bin].ranges.size() - 1].second) > 1)){
```
Comment (lomereiter) [inline src/algorithms/bin_path_info.cpp:75]: https://en.cppreference.com/w/cpp/container/vector/back

**Comment by lomereiter** [comment]
> Have you profiled the change in run speed and output size? I had almost identical code in odgi earlier and I stripped it out because it seemed we only needed sign post nucleotides (first, last, median), not every single one.

AFAIU this PR only changes a single range [first, last] to multiple, to account for loops which would have otherwise be lumped into [ranges[0].first, ranges[N].last]. It's not dumping "every single one".

File: src/algorithms/bin_path_info.cpp (hunk 1 above)
Comment (subwaystation) [reply src/algorithms/bin_path_info.cpp:75]: I was not aware of this, will change accordingly.

**Comment by subwaystation** [comment]
@dimatr If I merge this in, we would have to update the JSON parsing in CS.
So I am going to wait until we have a similar PR in CS to update synchronously.

**Comment by subwaystation** [comment]
We will need odgi_version = 11

**Comment by subwaystation** [comment]
Following @josiahseaman Version 12 bumps merging this in, too.

**Pull request merged**